frontend/src/components/query/query-builder/steps/filter/filter-condition-tag.tsx

```diff
@@ -0,0 +1,173 @@
+"use client";
+import { Icon, IconName, Popover, Tag, Text } from "@blueprintjs/core";
+import {
+  FilterCondition,
+  InferredSchemaColumn,
+  InferredSchemaColumnSchema,
+} from "@/definitions";
+import SquareIcon, { SquareIconSize } from "@/components/icon/square-icon";
+import { getFormattedDateStrings } from "@/utils/value-format";
+import { useTable } from "@/data/use-tables";
+import { useDarkMode } from "@/stores";
+
+interface FilterConditionTagProps {
+  condition: FilterCondition;
+  index: number;
+  removeable: boolean;
+  removeCondition: (index: number) => void;
+}
+
+export default function FilterConditionTag({
+  condition,
+  index,
+  removeable,
+  removeCondition,
+}: FilterConditionTagProps) {
+  const [darkMode] = useDarkMode();
+  const {
+    data: columnTable,
+    isLoading: isLoadingColumnTable,
+    error: columnTableError,
+  } = useTable(condition.column.table);
+
+  const {
+    data: valueTable,
+    isLoading: isLoadingValueTable,
+    error: valueTableError,
+  } = useTable(
+    condition.value?.table && condition.value.table !== "aggregate"
```

Comment:
This works, but I'd like us to think of a more elegant solution for these fields, potentially just a different name as "aggregate" is used for other descriptions